pilot/pkg/networking/core/v1alpha3/envoyfilter/listener_patch.go

```diff
@@ -512,6 +520,7 @@ func patchHTTPFilters(patchContext networking.EnvoyFilter_PatchContext,
 			if removedFilters.Contains(filter.Name) {
 				continue
 			}
+			tempArray = append(tempArray, filter)
```

Comment:
This is the actual fix. Not adding to tempArray. Rest of the changes are format changes.

Reply:
Can you add a test? Lack of ut cause this happen.